src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -660,19 +661,18 @@ safe_void_coroutine WindowEmperor::_dispatchCommandlineCurrentDesktop(winrt::Ter
             if (desktopId == currentDesktop && lastActivatedTime > max)
             {
                 max = lastActivatedTime;
-                mostRecentWeak = w;
+                mostRecent = w;
             }
         }
-    }
-
-    // GetVirtualDesktopId(), as current implemented, should always return on the main thread.
-    _assertIsMainThread();
 
-    const auto mostRecent = mostRecentWeak.lock();
-    auto window = mostRecent.get();
-
-    if (!window)

```

Comment:
@DHowett The fix is that this if-condition is gone. If the `currentDesktop` was acquired we _must_ yield a `window = nullptr` if no window exists on the current desktop. This if-condition however will assign the most recent window.